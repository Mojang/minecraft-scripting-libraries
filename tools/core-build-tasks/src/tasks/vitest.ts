import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
