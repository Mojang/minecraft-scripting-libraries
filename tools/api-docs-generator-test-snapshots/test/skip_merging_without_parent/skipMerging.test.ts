// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Skip Merging Without Parent', () => {
    it('Properly skips merging if skip merging is set to true', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['ts'],
            skipMerging: true,
        });
    });
});
