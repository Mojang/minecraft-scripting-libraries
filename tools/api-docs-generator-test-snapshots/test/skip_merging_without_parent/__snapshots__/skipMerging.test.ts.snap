// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skip Merging Without Parent > Properly skips merging if skip merging is set to true > # Of Files Generated 1`] = `1`;

exports[`Skip Merging Without Parent > Properly skips merging if skip merging is set to true > typescript/@minecraft/server@2.2.0-beta.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @beta
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "@minecraft/server",
 *   "version": "2.2.0-beta"
 * }
 * \`\`\`
 *
 */
/**
 * @remarks
 * This function can't be called in read-only mode.
 *
 */
export function ChildFunction(): number;
"
`;
